### PR TITLE
Changes Substitute method: set_reserved(value) -> set_reserved

### PR DESCRIPTION
**Pull request by arepnikov**

### CODE REVIEW

File: lib/idempotence/reservation/reservation.rb [hunk 1]
```diff
@@ -107,17 +107,22 @@ def self.build
...
 
-          yield message if @process_block
+          yield message if process_block
         end
 
-        def set_reserved(value)
-          @process_block = value
+        def process_block
+          @process_block ||= false
+        end
```
Comment (AlfonsoUceda) [inline lib/idempotence/reservation/reservation.rb:120]: We can have an attr_reader and initialize `process_block` to false in the initialize method of the substitute class, no need for this memoization.

File: lib/idempotence/reservation/reservation.rb [hunk 2]
```diff
@@ -107,17 +107,22 @@ def self.build
...
+        attr_writer :process_block
+
+        def set_reserved!
```
Comment (samstickland) [inline lib/idempotence/reservation/reservation.rb:123]: Not sure about the `!` suffix - why is it there?

File: lib/idempotence/reservation/reservation.rb (hunk 1 above)
Comment (samstickland) [reply lib/idempotence/reservation/reservation.rb:120]: Agreed, this method looks unnecessary. Unset instance variables are naturally falsey

**Review by samstickland — CHANGES_REQUESTED** [review]

File: lib/idempotence/reservation/reservation.rb (hunk 2 above)
Comment (AlfonsoUceda) [reply lib/idempotence/reservation/reservation.rb:123]: Yeah, I think there is no need for that. I understand this happens in ruby core or activerecord just to give another meaning, like modifying the data structure or raising  an exception

File: lib/idempotence/reservation/reservation.rb (hunk 1 above)
Comment (arepnikov) [reply lib/idempotence/reservation/reservation.rb:120]: agreed

File: lib/idempotence/reservation/reservation.rb (hunk 2 above)
Comment (AlfonsoUceda) [reply lib/idempotence/reservation/reservation.rb:123]: It's not common to use `set_` and `get_` in ruby but we do quite often in substitutes, e.g. `set_result` I think `set_reserved` is enough.

**Review by AlfonsoUceda — APPROVED** [review]
Just one change

**Review by samstickland — APPROVED** [review]

**Pull request merged**